server/moodle/mod/livequiz/classes/livequiz/livequiz.php

```diff
@@ -0,0 +1,247 @@
+        $this->timecreated = $timecreated;
+        $this->timemodified = $timemodified;
+
+        return $this;
+    }
+
+    /**
+     * Gets a livequiz instance from the DB, with all relevant attributes.
+     *
+     * TODO implement associated lecturer.
+     *
+     * @param int $id
+     * @return livequiz
+     * @throws dml_exception
+     */
+    public static function get_livequiz_instance(int $id): livequiz {
+        global $DB;
+        $quizinstance = $DB->get_record('livequiz', ['id' => $id]);
+
+        return new livequiz(
+            $quizinstance->id,
+            $quizinstance->name,
+            $quizinstance->course,
+            $quizinstance->intro,
+            $quizinstance->introformat,
+            $quizinstance->timecreated,
+            $quizinstance->timemodified
+        );
+    }
+
+    /**
+     * Updates the livequiz in the database, and updates the timemodified field.
+     *
+     * @param livequiz $livequiz
+     * @return bool
+     * @throws dml_exception
+     */
+    public function update_quiz(): bool {
+        global $DB;
+
+        $this->set_timemodified();
+
+        $record = new stdClass();
+        $record->id = $this->get_id();
+        $record->timemodified = $this->get_timemodified();
+
+        return $DB->update_record('livequiz', $record);
+    }
+
+    /**
+     * Gets the associated ID for the livequiz.
+     *
+     * @return int
+     */
+    public function get_id(): int {
+        return $this->id;
+    }
+
+    /**
+     * Gets the name of the livequiz.
+     *
+     * @return string
+     */
+    public function get_name(): string {
+        return $this->name;
+    }
+
+    /**
+     * Gets the ID for the assicated course
+     *
+     * @return int
+     */
+    public function get_course(): int {
+        return $this->course;
+    }
+
+    /**
+     * Gets the introduction for the livequiz.
+     *
+     * @return string
+     */
+    public function get_intro(): string {
+        return $this->intro;
+    }
+
+    /**
+     * Gets the time the livequiz was created.
+     *
+     * @return int
+     */
+    public function get_timecreated(): int {
+        return $this->timecreated;
+    }
+
+    /**
+     * Gets the time the livequiz was last modified.
+     *
+     * @return int
+     */
+    public function get_timemodified(): int {
+        return $this->timemodified;
+    }
+
+    /**
+     * Gets the questions associated with the livequiz.
+     *
+     * @return array
+     */
+    public function get_questions(): array {
+        return $this->questions;
+    }
+
+
+    /**
+     * Sets the timemodified field to the current time.
+     *
+     * @return int
+     */
+    public function set_timemodified(): int {
+        return $this->timemodified = time();
+    }
+
+    /**
+     * Sets the questions associated with the livequiz.
+     *
+     * @param array $questions
+     */
+    public function set_questions(array $questions): void {
```

Comment:
Should be named add_questions. Also comment should be changed to appends instead of sets